library/common.py

```diff
@@ -5,25 +5,120 @@
 from urllib.parse import urljoin
 from urllib.parse import quote
 
-class ProvisionerError(Exception):
-    def __init__(self, message):
-        super(ProvisionerError, self).__init__(message)
-
 # TODO: This needs a more standard name (static method)
 # Or we need to make sure the IPGetter object is easy to use
-def get_machine_by_name(mrp_token, mrp_url, machine_name):
+def get_machine_by_name(requester, machine_name):
```

Comment:
Nit: "requester" is a horrible name. :)

Something like "URLhandler" would be better.

Reply:
It sounds charismatic ! Bah, changed it to URLhandler

Reply:
Please, also change all "requester" variables and arguments to "handler".